src/validation.cpp

```diff
@@ -2867,7 +2875,8 @@ bool CChainState::ActivateBestChain(BlockValidationState &state, const CChainPar
         LimitValidationInterfaceQueue();
 
         {
-            LOCK2(cs_main, ::mempool.cs); // Lock transaction pool for at least as long as it takes for connectTrace to be consumed
```

Comment:
We want to ditch LOCK2?

Reply:
No it can stay, but here `LOCK2` doesn't support pointers or interfaces, so if someone wants to pick up #19629 again, `LOCK(cs_main)` will likely end up to be separate anyway.